torch/lib/THD/base/init_methods/InitMethodTCP.cpp

```diff
@@ -35,8 +35,11 @@ std::string getRandomString()
 
   int fd;
   unsigned int seed;
+  ssize_t bytes_read;
   SYSCHECK(fd = open("/dev/urandom", O_RDONLY));
-  SYSCHECK(read(fd, &seed, sizeof(seed)));
+  SYSCHECK(bytes_read = read(fd, &seed, sizeof(seed)));
```

Comment:
I forgot how messed up the C++11 random library is.

What's the point of the `uid` here? It seems like you already have uniqueness with the group_name and rank.

If that's not sufficient, then why not just use the single value you get from `/dev/urandom`? There's no additional randomness (or uniqueness) from passing it through the Mersenne twister or the encoding it into the vocabulary.